gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dweaponpreview_ttt2.lua

```diff
@@ -0,0 +1,377 @@
+---
+-- @class PANEL
+-- @section DWeaponPreviewTTT2
+
+local mathMax = math.max
+local mathMin = math.min
+
+local PANEL = {}
+
+---
+-- @accessor number
+-- @realm client
+AccessorFunc(PANEL, "m_fAnimSpeed", "AnimSpeed")
+
+---
+-- @accessor Entity
+-- @realm client
+AccessorFunc(PANEL, "Entity", "Entity")
+
+---
+-- @accessor Vector
+-- @realm client
+AccessorFunc(PANEL, "vCamPos", "CamPos")
+
+---
+-- @accessor number
+-- @realm client
+AccessorFunc(PANEL, "fFOV", "FOV")
+
+---
+-- @accessor Vector
+-- @realm client
+AccessorFunc(PANEL, "vLookatPos", "LookAt")
+
+---
+-- @accessor Angle
+-- @realm client
+AccessorFunc(PANEL, "aLookAngle", "LookAng")
+
+---
+-- @accessor Color
+-- @realm client
+AccessorFunc(PANEL, "colAmbientLight", "AmbientLight")
+
+---
+-- @accessor Color
+-- @realm client
+AccessorFunc(PANEL, "colColor", "Color")
+
+---
+-- @accessor boolean
+-- @realm client
+AccessorFunc(PANEL, "bAnimated", "Animated")
+
+---
+-- @accessor boolean
+-- @realm client
+AccessorFunc(PANEL, "bHoverEffect", "HoverEffect")
+
+---
+-- @ignore
+function PANEL:Init()
+	self.lastPaint = 0
+	self.directionalLight = {}
+	self.farZ = 4096
+
+	self:SetContentAlignment(5)
+
+	self:SetTall(22)
+	self:SetMouseInputEnabled(true)
+	self:SetKeyboardInputEnabled(true)
+
+	self:SetFont("DermaTTT2TextLarge")
+
+	self:SetCamPos(Vector(75, -65, 55))
+	self:SetLookAt(Vector(5, 0, 35))
+	self:SetFOV(35)
+
+	self:SetAnimSpeed(0.5)
+	self:SetAnimated(true)
+	self:SetHoverEffect(true)
+
+	self:SetAmbientLight(Color(150, 150, 150))
+
+	self:SetDirectionalLight(BOX_TOP, Color(255, 255, 255))
+	self:SetDirectionalLight(BOX_FRONT, Color(255, 255, 255))
+
+	self:SetColor(COLOR_WHITE)
+
+	-- remove label and overwrite function
+	self:SetText("")
+
+	self.data = {
+		ply = nil,
+		wep = nil,
+		wepClass = "",
+		HoldType = "normal",
+		worldModelData = {},
+		worldModels = {},
+		specialModelHandeling = false
+	}
+end
+
+---
+-- @realm client
+function PANEL:OnRemove()
+	-- old ent is removed because clientside models are not garbage collected
+	if IsValid(self.data.ply) then
+		self.data.ply:Remove()
+	end
+	if IsValid(self.data.wep) then
+		self.data.wep:Remove()
+	end
+end
+
+---
+-- @param number iDirections
```

Comment:
where does this number come from?

Reply:
![image](https://github.com/TTT-2/TTT2/assets/13639408/c3e548a0-df80-4e77-92ab-9444c65fb036)

This is taken from the other 3d model UI renderers in Gmod

Reply:
Yes but the documentation for this parameter could mention the ENUM this is coming from instead of `iDirections`